### PR TITLE
 GH-1676: only open rhs when in a channel

### PR DESCRIPTION
**Pull request by calebroseland**
# Summary
Adds a check to ensure we're viewing a channel before opening RHS. `currentChannelId` can sometimes have a value even though you're not in a channel. 

# Ticket Link
- Fixes: #1676 

## Checklist

- ~[ ] Telemetry updated~
- ~[ ] Gated by experimental feature flag~
- [x] Unit tests updated

### CODE REVIEW

**Review by trilopin — APPROVED** [review]
Thanks 🎉

File: webapp/src/rhs_opener.ts [hunk 1]
```diff
@@ -24,12 +26,14 @@ export function makeRHSOpener(store: Store<GlobalState>): () => Promise<void> {
         const currentChannel = getCurrentChannel(state);
         const currentTeam = getCurrentTeam(state);
         const playbookRun = currentPlaybookRun(state);
+        const url = new URL(window.location.href);
+        const isInChannel = matchPath(url.pathname, {path: '/:team/:path(channels|messages)/:identifier/:postid?'});
```
Comment (lieut-data) [inline webapp/src/rhs_opener.ts:30]: Do these parameters implicitly have to match whatever was registered by the webapp? (i.e. are these brittle)

Wondering if it makes sense to just verify that we're in the Channels product instead?

**Review by lieut-data — APPROVED** [review]

**Comment by lieut-data** [comment]
Also: any hope of an E2E test?

**Comment by calebroseland** [comment]
> Also: any hope of an E2E test?

Yup, was going to try to get back to this for that today

File: webapp/src/rhs_opener.ts (hunk 1 above)
Comment (calebroseland) [reply webapp/src/rhs_opener.ts:30]: This is indeed brittle, but unavoidable to my understanding. All of the `currentProduct` and like selectors are all exclusively in mm-webapp, and it'd be just as brittle IMO to raw-select the products and manually recreate `useCurrentProductId`. I'm only 1/5 on the current path-check; happy to switch it over.

**Pull request merged**